backend/recommendation.js

```diff
@@ -0,0 +1,91 @@
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+
+function generateTimeGrid() {
+    const timeGrid = [];
+    for (let i = 0; i < 48; i++) {
+        timeGrid.push({ slot: i, available: true });
+    }
+    return timeGrid;
+}
+
+function markUnavailableSlots(timeGrid, startAt, endAt) {
+    const startSlot = Math.floor(startAt.getHours() * 2 + startAt.getMinutes() / 30);
```

Comment:
a comment here may help understand this part. From what I understand, each hour takes up two slots in your time grid, and every 30 minutes take up 1 slot?